### PR TITLE
RobotLog: Add README and diagram

### PR DESCRIPTION
**Pull request by coderkevin**
This adds a README file for RobotLog to describe its features and
operation, and adds a diagram for visual explanation.

To view README, click here: https://github.com/frc2357/robotlib2357/blob/add/robot-log-docs/docs/robot-log/README.md

### CODE REVIEW

File: docs/robot-log/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	LogOutput stdOut = new PrintStreamLogOutput(this, System.out);
+
+	// Write errors to stderr output (viewable in driver station)
+	LogOutput stdErr = new PrintStreamLogOutput(this, System.err);
+
+	// Write to files in /home/lvuser/logs/competition-log-##.zip
+	LogOutput logFile = new FileLogOutput(this, "~/logs", "competition-log-");
+
+	public void onStart() {
+		subscribeTopic("drive-error", stdErr);
```
Comment (shadwstalkr) [inline docs/robot-log/README.md]: This doesn't match the interface in your later patch.

File: docs/robot-log/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	// Write informational messages to output (viewable in driver station)
+	LogOutput stdOut = new PrintStreamLogOutput(this, System.out);
+
+	// Write errors to stderr output (viewable in driver station)
+	LogOutput stdErr = new PrintStreamLogOutput(this, System.err);
+
+	// Write to files in /home/lvuser/logs/competition-log-##.zip
+	LogOutput logFile = new FileLogOutput(this, "~/logs", "competition-log-");
+
+	public void onStart() {
```
Comment (shadwstalkr) [inline docs/robot-log/README.md]: Does LogSession need to call a protected onStart method in its' start method?

File: docs/robot-log/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+### Example of a competition match logging session
+
+Use LogSession to create outputs and link them to topics.
+
+```
+public class CompetitionLogSession extends LogSession {
+	// Write informational messages to output (viewable in driver station)
+	LogOutput stdOut = new PrintStreamLogOutput(this, System.out);
```
[the remark this answers is not in the document]
Comment (coderkevin) [reply docs/robot-log/README.md]: Yes it is, it's basically an extension of the constructor.

File: docs/robot-log/README.md (hunk 2 above)
Comment (coderkevin) [reply docs/robot-log/README.md]: It does. I added it and will update this to be protected as well. Thanks!

File: docs/robot-log/README.md (hunk 1 above)
Comment (coderkevin) [reply docs/robot-log/README.md]: You're right. I didn't change it yet. I've added a commit for that on the LogSession PR. Thanks!

**Pull request merged**